app/src/main/java/pm/gnosis/heimdall/ui/settings/general/GeneralSettingsViewModel.kt

```diff
@@ -19,5 +22,16 @@ class GeneralSettingsViewModel @Inject constructor(
             .mapToResult()
 
     override fun loadPaymentToken() =
-            tokenRepository.loadPaymentToken()
+        tokenRepository.loadPaymentToken()
+
+    override fun loadSafeAdresses(): Single<List<String>> =
+        safeRepository.observeAllSafes()
+            .map {
+                val addresses = mutableListOf<String>()
+                it.forEach { safe ->
+                    addresses.add(safe.address().value.toString(16).addHexPrefix())
```

Comment:
there is a method `.asEthereumAddressChecksumString()` that generates a checksumed ethereum address string with prefix

Reply:
done